### PR TITLE
feat(live-preview): adding live preview to the template [TOL-1081] 

### PR DESCRIPTION
**Pull request by chrishelgert**
## Purpose of PR

* Cherry-Pick #193 into main-private
  * Adding live preview SDK with inspector mode (field tagging) and live updates
  * remove X-Ray
  * Configure security headers for live preview
  * Show the settings panel not inside the live preview iframe

## Testing

* Locally it should be everything the same (except x-ray is removed)
* If used inside live preview (configured preview environments)
  * you can enable the inspector mode and see the tagged fields
  * clicking on them opens the editor
  * changing something reflects also directly on the website
  * the button for the settings is not shown

## Conflicted files

* next.config.js
* yarn.lock
* settings.tsx
* settings-form.tsx

### CODE REVIEW

File: src/_ctf-private/ctf-analytics/CtfSegmentAnalytics.tsx [hunk 1]
```diff
@@ -26,12 +26,11 @@ export const CtfSegmentAnalytics = () => {
     if (initialized && analytics) {
       analytics.page({
         templateId: 'marketing',
-        xRayActive: Boolean(xray),
         previewActive: Boolean(preview),
         guestSpaceActive: Boolean(guestSpaceActive),
       });
     }
-  }, [guestSpaceActive, initialized, preview, xray]);
+  }, [guestSpaceActive, initialized, preview]);
```
Comment (mattvanvoorst-contentful) [inline src/_ctf-private/ctf-analytics/CtfSegmentAnalytics.tsx:33]: Once we remove xray, we should notify Rosi and update the Segment tracking plan for starter-templates accordingly

Comment (chrishelgert) [reply src/_ctf-private/ctf-analytics/CtfSegmentAnalytics.tsx:33]: we can only adjust the segment plan once we've adjusted this for every starter-template?

Comment (mattvanvoorst-contentful) [reply src/_ctf-private/ctf-analytics/CtfSegmentAnalytics.tsx:33]: That's a good shout - hadn't considered that. Alright for now we should be okay to ignore it; I think Segment still sends the end

Comment (chrishelgert) [reply src/_ctf-private/ctf-analytics/CtfSegmentAnalytics.tsx:33]: Ok I add it to the follow up tickets for the remaining starter templates

Comment (cf-remylenoir) [reply src/_ctf-private/ctf-analytics/CtfSegmentAnalytics.tsx:33]: For reference, this is the [document](https://contentful.atlassian.net/wiki/spaces/PROD/pages/4138500311/Starter+Templates+-+Segment+tracking#Tracking-plan) that needs to be updated once the event is removed from the Tracking plan

**Pull request merged**